tests/platform_tests/fwutil/conftest.py

```diff
@@ -0,0 +1,107 @@
+import tarfile
+import json
+import pytest
+import logging
+
+from random import randrange
+
+from fwutil_common import show_firmware
+
+logger = logging.getLogger(__name__)
+
+DUT_HOME="/home/admin"
+DEVICES_PATH="/usr/share/sonic/device"
+FS_PATH_TEMPLATE = "/host/image-{}/fs.squashfs"
+FS_RW_TEMPLATE = "/host/image-{}/rw"
+FS_WORK_TEMPLATE = "/host/image-{}/work"
+FS_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-fs"
+OVERLAY_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-overlay"
+
+def check_path_exists(path):
+    return duthost.stat(path = path)["stat"]["exists"] 
+
+def pytest_generate_tests(metafunc):
+    val = metafunc.config.getoption('--fw-pkg')
+    if 'fw_pkg_name' in metafunc.fixturenames and val is not None:
+        metafunc.parametrize('fw_pkg_name', [val], scope="module")
+
+@pytest.fixture(scope='module')
+def fw_pkg(fw_pkg_name):
+    logger.info("Unpacking firmware package to ./firmware")
+    os.mkdir("firmware")
```

Comment:
@alexrallen how about https://docs.python.org/3/library/tempfile.html ?

Reply:
Passing around a dynamic directory name would clutter the code. This is a very trivial use case here as such we don't need a  library to handle it. Its more likely for bugs to be created by subtle behavior of this system than it is for the manual creation of a directory in python.

Reply:
@alexrallen what if there is bug and package doesn't have a config file? In that case, it will extract on top of previous version and will fail with an error.

Reply:
Everything is contained in firmware.tar.gz there is only one extraction that occurs per run of this module. If there is a stale firmware folder the new one will unpack on top of it and overwrite it. The likelihood of 

1. Uploading a new firmware.tar.gz without a firmware.json file 
2. Having a stale firmware folder present given our architecture (it refreshes the workspace at each new regression) 

is so low I don't think it justifies making a change here.